tasks/go_test.py

```diff
@@ -632,8 +632,13 @@ def test(
 
     stdlib_build_cmd = 'go build {verbose} -mod={go_mod} -tags "{go_build_tags}" -gcflags="{gcflags}" '
     stdlib_build_cmd += '-ldflags="{ldflags}" {build_cpus} {race_opt} std cmd'
-    cmd = 'gotestsum {junit_file_flag} {json_flag} --format pkgname {rerun_fails} --packages="{packages}" -- {verbose} -mod={go_mod} -vet=off -timeout {timeout}s -tags "{go_build_tags}" -gcflags="{gcflags}" '
-    cmd += '-ldflags="{ldflags}" {build_cpus} {race_opt} -short {covermode_opt} {coverprofile} {nocache} {test_run_arg}'
+    gotestsum_flags = '{junit_file_flag} {json_flag} --format pkgname {rerun_fails} --packages="{packages}"'
+    gobuild_flags = (
+        '-mod={go_mod} -tags "{go_build_tags}" -gcflags="{gcflags}" -ldflags="{ldflags}" {build_cpus} {race_opt}'
+    )
+    govet_flags = '-vet=off'
+    gotest_flags = '{verbose} -timeout {timeout}s -short {covermode_opt} {coverprofile} {test_run_arg} {nocache}'
+    cmd = f'gotestsum {gotestsum_flags} {gobuild_flags} {govet_flags} {gotest_flags}'
```

Comment:
I think we need a `--` between gotestsum flags and go test flags. If it works without it feel free to ignore
```suggestion
    cmd = f'gotestsum {gotestsum_flags} -- {gobuild_flags} {govet_flags} {gotest_flags}'
```

Reply:
You're right, just fixed it in [43ca303](https://github.com/DataDog/datadog-agent/pull/21918/commits/43ca3036964257270b6b90eedfc9f57f3a1b42fb), thanks !